core/common/iterator/src/main/java/org/eclipse/rdf4j/common/iteration/CloseableIteration.java

```diff
@@ -29,13 +31,9 @@
  * }
  * </pre>
  *
- * @deprecated In the future this interface will stop extending {@link Iteration} and instead declare the same interface
- *             methods directly. The interface will also stop requiring implementations to automatically close when
- *             exhausted, instead making this an optional feature and requiring the user to always call close. This
- *             interface may also be removed.
  */
-@Deprecated(since = "4.1.0")
-public interface CloseableIteration<E, X extends Exception> extends Iteration<E, X>, AutoCloseable {
+//@Deprecated(since = "4.1.0")
+public interface CloseableIteration<E> extends AutoCloseable, Iterator<E> {
```

Comment:
Removed the support for specifying exceptions with generics. We don't use this for checked exceptions anymore, so we're not loosing anything here.

We can now extend Iterator directly. 

One question is if we should rename this class to CloseableIterator and rename all use of "iteration" to "iterator". Since this class is used in so many places we could do a _soft_ rename and keep the old CloseableIteration around for a while more, but have it extend CloseableIterator.

Reply:
In favor of renaming. It's a big refactor whichever way we go, I'd rather get it in a clean/good state in one attempt.